Enable macos nightly build with Qt6.5

I recently tried publishing our nightly builds with the newly released Qt6.5 and hit a blocker with the macos build, see https://github.com/simulationcraft/simc-publish/pull/20

There are errors on macos with `std::filesystem::path`, see https://github.com/simulationcraft/simc-publish/actions/runs/4636357824/jobs/8204171694

```
/Users/runner/work/simc-publish/Qt/6.5.0/macos/lib/QtCore.framework/Headers/qfileinfo.h:99:22: error: 'path' is unavailable: introduced in macOS 10.15
    std::filesystem::path filesystemPath() const { return QtPrivate::toFilesystemPath(path()); }
                     ^
/Applications/Xcode_14.2.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX13.1.sdk/usr/include/c++/v1/filesystem:909:24: note: 'path' has been explicitly marked unavailable here
class _LIBCPP_TYPE_VIS path {
                       ^
In file included from ../qt/ImportTab.cpp:7:
In file included from ../qt/sc_AddonImportTab.hpp:8:
In file included from ../qt/util/sc_textedit.hpp:10:
In file included from /Users/runner/work/simc-publish/Qt/6.5.0/macos/lib/QtGui.framework/Headers/QtGui:3:
In file included from /Users/runner/work/simc-publish/Qt/6.5.0/macos/lib/QtGui.framework/Headers/QtGuiDepends:3:
In file included from /Users/runner/work/simc-publish/Qt/6.5.0/macos/lib/QtCore.framework/Headers/QtCore:80:
In file included from ../../Qt/6.5.0/macos/lib/QtCore.framework/Headers/qdir.h:9:
/Users/runner/work/simc-publish/Qt/6.5.0/macos/lib/QtCore.framework/Headers/qfileinfo.h:100:22: error: 'path' is unavailable: introduced in macOS 10.15
    std::filesystem::path filesystemAbsolutePath() const
                     ^
```

If anyone is interested and willing to fix this, please create a pull request for it. I assume that either adjusting some setting (macOS>= 10.15) or adjusting the libc++ version used might fix it, but I personally have no clue about how macos works :-)

fyi @navv1234 @wfarr @philipp-spiess @nuoHep

--- a/qt/CMakeLists.txt
+++ b/qt/CMakeLists.txt
@@ -31,6 +31,7 @@ endif()
 add_executable(SimulationCraft WIN32 ${source_files})
 
 if (${QT_VERSION_MAJOR} EQUAL 6)
+  # std::filesystem (used indirectly with Qt6.5 on newer compilers) usage requires minimum osx target version of 10.15
   set(CMAKE_OSX_DEPLOYMENT_TARGET "10.15")
 endif()
 

--- a/qt/CMakeLists.txt
+++ b/qt/CMakeLists.txt
@@ -30,6 +30,10 @@ if (WIN32)
 endif()
 add_executable(SimulationCraft WIN32 ${source_files})
 
+if (${QT_VERSION_MAJOR} EQUAL 6)
+  set(CMAKE_OSX_DEPLOYMENT_TARGET "10.15")
+endif()
+
 sc_common_compiler_options(SimulationCraft)
 
 target_include_directories(SimulationCraft PUBLIC ../engine/)

